clouddriver-web/src/test/groovy/com/netflix/spinnaker/clouddriver/controllers/CredentialsControllerSpec.groovy

```diff
@@ -58,6 +59,54 @@ class CredentialsControllerSpec extends Specification {
     parsedResponse == [[name: "test", environment: "env", accountType: "acctType", cloudProvider: "testProvider", type: "testProvider", requiredGroupMembership: ["test"], permissions: [READ:["test"], WRITE:["test"]], challengeDestructiveActions: false, primaryAccount: false]]
   }
 
+  void "credentials are listed by type"() {
+    setup:
+
+    def objectMapper = new ObjectMapper().setSerializationInclusion(JsonInclude.Include.NON_NULL)
+    def credsRepo = new MapBackedAccountCredentialsRepository()
+    def accountDefRepo = new MapBackedAccountDefinitionRepository()
+    def credsProvider = new DefaultAccountCredentialsProvider(credsRepo)
+    def accountDefSrvc = Optional.of(new DefaultAccountDefinitionService(accountDefRepo))
+    accountDefRepo.save(new ManagedAccount("test1", "acctType1"))
+    accountDefRepo.save(new ManagedAccount("test2", "acctType2"))
+    accountDefRepo.save(new ManagedAccount("test3", "acctType3"))
+    def mvc = MockMvcBuilders.standaloneSetup(new CredentialsController(accountDefSrvc, new CredentialsConfiguration(), objectMapper, credsProvider)).build()
+
+    when:
+    def result = mvc.perform(MockMvcRequestBuilders.get("/credentials/type/acctType1").accept(MediaType.APPLICATION_JSON)).andReturn()
```

Comment:
The important thing here is that either limit or account name or both aren't present, right?  If so, then please add this to the description of the test.

Reply:
Fixed.